# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Element.h
# Commit: f9cf0740eef2
# Full Hash: f9cf0740eef2fcae3e2f93afe49df1569bd89c03
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 37e2a10a9022 -r f9cf0740eef2 dom/base/Element.h
--- a/dom/base/Element.h	Tue May 14 16:43:27 2024 +0000
+++ b/dom/base/Element.h	Tue May 14 16:43:28 2024 +0000
@@ -120,6 +120,7 @@
 struct CheckVisibilityOptions;
 struct CustomElementData;
 struct SetHTMLOptions;
+struct GetHTMLOptions;
 struct GetAnimationsOptions;
 struct ScrollIntoViewOptions;
 struct ScrollToOptions;
@@ -1359,11 +1360,13 @@
 
   enum class DelegatesFocus : bool { No, Yes };
   enum class ShadowRootClonable : bool { No, Yes };
+  enum class ShadowRootSerializable : bool { No, Yes };
 
   already_AddRefed<ShadowRoot> AttachShadowWithoutNameChecks(
       ShadowRootMode aMode, DelegatesFocus = DelegatesFocus::No,
       SlotAssignmentMode aSlotAssignmentMode = SlotAssignmentMode::Named,
-      ShadowRootClonable aClonable = ShadowRootClonable::No);
+      ShadowRootClonable aClonable = ShadowRootClonable::No,
+      ShadowRootSerializable aSerializable = ShadowRootSerializable::No);
 
   // Attach UA Shadow Root if it is not attached.
   enum class NotifyUAWidgetSetup : bool { No, Yes };
@@ -1543,6 +1546,7 @@
 
   void SetHTML(const nsAString& aInnerHTML, const SetHTMLOptions& aOptions,
                ErrorResult& aError);
+  void GetHTML(const GetHTMLOptions& aOptions, nsAString& aResult);
 
   //----------------------------------------
 